habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -139,6 +139,140 @@ def get_ao_global_bb(
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
+
+    :return: tuple (local_aabb, global_transform)
+    """
+
+    # stage bounding box
+    if obj_id == habitat_sim.stage_id:
+        return (
+            sim.get_active_scene_graph().get_root_node().cumulative_bb,
+            mn.Matrix4.identity_init(),
+        )
+
+    obj = get_obj_from_id(sim, obj_id, ao_link_map)
+
+    assert (
+        obj is not None
+    ), f"object id {obj_id} is not found, this is unexpected. Invalid/stale object id?"
+
+    if isinstance(obj, habitat_sim.physics.ManagedRigidObject):
+        return (obj.root_scene_node.cumulative_bb, obj.transformation)
+
+    # ManagedArticulatedObject
+    if obj.object_id == obj_id:
+        # this is the AO itself
+        ao_aabb = None
+        if ao_aabbs is None or obj_id not in ao_aabbs:
+            ao_aabb = get_ao_root_bb(obj)
+        else:
+            ao_aabb = ao_aabbs[obj_id]
+        return (ao_aabb, obj.transformation)
+
+    # this is a link
+    link_node = obj.get_link_scene_node(obj.link_object_ids[obj_id])
+    link_transform = link_node.absolute_transformation()
+    return (link_node.cumulative_bb, link_transform)
+
+
+def get_obj_size_along(
+    sim: habitat_sim.Simulator,
+    object_id: int,
+    global_vec: mn.Vector3,
+    ao_link_map: Dict[int, int] = None,
+    ao_aabbs: Dict[int, mn.Range3D] = None,
+) -> Tuple[float, mn.Vector3]:
+    """
+    Uses object bounding box ellipsoid scale as a heuristic to estimate object size in a particular global direction.
+
+    :param sim: The Simulator instance.
+    :param object_id: The integer id of the object or link.
+    :param global_vec: Vector in global space indicating the direction to approximate object size.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
+
+    :return: distance along the specified direction and global center of bounding box from which distance was estimated.
+    """
+
+    obj_bb, transform = get_bb_for_object_id(
+        sim, object_id, ao_link_map, ao_aabbs
+    )
+    center = transform.transform_point(obj_bb.center())
+    local_scale = mn.Matrix4.scaling(obj_bb.size() / 2.0)
+    local_vec = transform.inverted().transform_vector(global_vec).normalized()
+    local_vec_size = local_scale.transform_vector(local_vec).length()
+    return local_vec_size, center
+
+
+def size_regularized_distance(
+    sim: habitat_sim.Simulator,
+    object_id_a: int,
+    object_id_b: int,
+    ao_link_map: Dict[int, int] = None,
+    ao_aabbs: Dict[int, mn.Range3D] = None,
+) -> float:
+    """
+    Get the heuristic surface-to-surface distance between two objects (regularized by their individual heuristic sizes).
```

Comment:
this is not surface-to-surface but more center-to-center since its the center of the bouinding box?

Reply:
Ah I see the final return which is from surface to surface.